presto-main/src/main/java/io/prestosql/operator/PageWithPositionEqualsAndHash.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import io.prestosql.spi.Page;
+
+/**
+ * Equals and hash evaluated at the specified page positions. Implementations need to be hash table friendly.
+ *
+ * Note: this usage is likely to be megamorphic, so use at your own risk!
+ */
+public interface PageWithPositionEqualsAndHash
```

Comment:
Could we limit scope of this interface to just `PagesWithPositionHash`? `PageWithPositionComparator` already deals with equality

Reply:
There are more optimizations that exist when doing equals than compare (such as variable length values). Given that the hash lookups are the hotspot in this code, it makes sense to give it every chance for speed up -- just as in many ways java objects still implement equals when there is a compareTo available.